src/gwt/src/org/rstudio/studio/client/application/ui/CommandPalette.java

```diff
@@ -0,0 +1,442 @@
+import com.google.gwt.user.client.ui.TextBox;
+import com.google.gwt.user.client.ui.Widget;
+
+import elemental.events.KeyboardEvent.KeyCode;
+
+/**
+ * CommandPalette is a widget that displays all available RStudio commands in a
+ * searchable list.
+ */
+public class CommandPalette extends Composite
+{
+   private static CommandPaletteUiBinder uiBinder = GWT.create(CommandPaletteUiBinder.class);
+
+   interface CommandPaletteUiBinder extends UiBinder<Widget, CommandPalette>
+   {
+   }
+   
+   /**
+    * The host interface represents the class hosting the widget (not a widget
+    * itself), which is currently the CommandPaletteLauncher.
+    */
+   public interface Host
+   {
+      public void dismiss();
+   }
+   
+   public interface Styles extends CssResource
+   {
+      String popup();
+      String searchBox();
+      String commandList();
+      String commandPanel();
+   }
+
+   public CommandPalette(Commands commands, RAddins addins, ShortcutManager shortcuts, Host host)
+   {
+      initWidget(uiBinder.createAndBindUi(this));
+
+      entries_ = new ArrayList<CommandPaletteEntry>();
+      host_ = host;
+      shortcuts_ = shortcuts;
+      selected_ = -1;
+      addins_ = addins;
+      commands_ = commands;
+      styles_.ensureInjected();
+      
+      Element searchBox = searchBox_.getElement();
+      searchBox.setAttribute("placeholder", "Search and run commands");
+      searchBox.setAttribute("spellcheck", "false");
+
+      // Accessibility attributes: list box
+      Element commandList = commandList_.getElement();
+      ElementIds.assignElementId(commandList, ElementIds.COMMAND_PALETTE_LIST);
+      Roles.getListboxRole().set(commandList);
+      Roles.getListboxRole().setAriaLabelProperty(commandList, "Matching commands");
+
+      // Accessibility attributes: search box
+      ElementIds.assignElementId(searchBox_, ElementIds.COMMAND_PALETTE_SEARCH);
+      Roles.getComboboxRole().setAriaOwnsProperty(searchBox, Id.of(commandList_.getElement()));
+      Roles.getComboboxRole().set(searchBox);
+      Roles.getComboboxRole().setAriaLabelProperty(searchBox, "Search for and run a command");
+      Roles.getComboboxRole().setAriaExpandedState(searchBox, ExpandedValue.TRUE);
+      A11y.setARIAAutocomplete(searchBox_, "list");
+      
+      // Accessibility attributes: announcement region; we want this to be read
+      // when filter updates are complete
+      A11y.setVisuallyHidden(resultsCount_);
+      Roles.getAlertRole().setAriaLiveProperty(resultsCount_.getElement(), LiveValue.ASSERTIVE);
+
+      // Populate the palette on a deferred callback so that it appears immediately
+      Scheduler.get().scheduleDeferred(() ->
+      {
+         populate();
+      });
+      
+      // Debounced update of the result count for screen readers; we debounce
+      // this so that every keystroke doesn't trigger an announcement.
+      updateResultsCount_ = new DebouncedCommand(1000)
```

Comment:
We have a user pref for this value: `typingStatusDelayMs()`

Reply:
Thanks, this is now respected!